packages/core/botpress/src/dialog/engine.js

```diff
@@ -664,6 +665,7 @@ bp.dialogEngine.onBeforeSessionTimeout((ctx, next) => {
   async _invokeAction(instruction, userState, event, context) {
     let name = null
     let args = {}
+    const hasVariable = str => /{{\/?[^}}]+}}/g.test(str)
```

Comment:
In processors I think we have a simpler check, just `if (rendered && rendered.includes('{{')) {`, maybe it's good enough here as well?

Reply:
I use this regex for exclude, something like that `text{{vartext`. Do mustache predict this behavior?

Reply:
I tested variante `text{{var`. In result I had follow 
```error: [notification::botpress] DialogEngine: ERROR function has invalid arguments (not a valid JSON string): {"name":"value","age":"/new-test{{event.text"}```

Reply:
@emirotin @epaminond @slvnperron  how you watch on variant to add validation for input? Maybe it will be one of best solution)

Reply:
@dmytropanontko , not sure we need to have input validation - it may appear that you want to use curly braces for something in text - e.g. for emoticons like `<.{{{><`. I think regexp check before rendering would be fine.

Reply:
@emirotin , so what do we do here? ))

Reply:
I think it should be a very simple check (for performance only), and then _try_ to mustache the string and catch the error (using the original string if mustaching fails)